### PR TITLE
Feature/oe 259 stock points

### PR DESCRIPTION
**Pull request by ron-junior-devsquad**
## Review

For the submitter, initial self-review:

- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency between the changeset and the goal stated above
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Complexity - is the proposed change introducing any unjustified complexity? 
- [ ] Thoroughness of added tests and any missing edge cases

### CODE REVIEW

File: app/Models/User.php [hunk 1]
```diff
@@ -435,6 +446,32 @@ public function hasRole(string $role): bool
...
+                                  })->with('stockPoint')->get()->sum(fn($customer) => $customer->stockPoint->stock_department)) +
+                                  ($this->customersManagedRegion()->whereHas('stockPoint', function ($query) {
+                                      $query->whereYear('created_at', Carbon::now());
+                                  })->with('stockPoint')->get()->sum(fn($customer) => $customer->stockPoint->stock_regional)) +
+                                  $this->customersManagedOffice()->whereHas('stockPoint', function ($query) {
+                                      $query->whereYear('created_at', Carbon::now());
+                                  })->with('stockPoint')->get()->sum(fn($customer) => $customer->stockPoint->stock_manager) 
+
+        ];
+    }
```
Comment (devsquad-mathews-lima) [inline app/Models/User.php:473]: Idk man, but IMHO, this is a messy code :sweat_smile:

Maybe break in smaller functions and just make a sum of it? Something like

```
    $this->sumOfSalesRepStockPersonalSafe() + ...
```

File: database/seeders/MultiplierOfYearSeeder.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Database\Seeders;
+
+use App\Models\MultiplierOfYear;
+use Illuminate\Database\Seeder;
+
+class MultiplierOfYearSeeder extends Seeder
```
Comment (devsquad-mathews-lima) [inline database/seeders/MultiplierOfYearSeeder.php:8]: Is necessary to put this seeder in `DatabaseSeeder.php`?

When I enter in homepage as an Admin, I get an error.

image

**Review by devsquad-mathews-lima — CHANGES_REQUESTED** [review]
When I was trying to edit a **customer**, this error showed up

image

File: tests/Feature/Profile/StockPointsTest.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            $this->teamStockPoints[0]->stock_setting + 
+            $this->teamStockPoints[1]->stock_recruiter + 
+            $this->teamStockPoints[2]->stock_manager +
+            $this->teamStockPoints[3]->stock_regional +
+            $this->teamStockPoints[4]->stock_department
+        );
+    }
+
+    /** @test */
+    public function it_should_show_gross_stock_points()
```
[the remark this answers is not in the document]
Comment (ron-junior-devsquad) [reply tests/Feature/Profile/StockPointsTest.php:125]: fixed, this test must validate if numbe is as int but the result did comming as a float

File: app/Models/User.php (hunk 1 above)
Comment (ron-junior-devsquad) [reply app/Models/User.php:473]: So i pushed a new code. if you think that still not good. tell me pls

File: database/seeders/MultiplierOfYearSeeder.php (hunk 2 above)
Comment (ron-junior-devsquad) [reply database/seeders/MultiplierOfYearSeeder.php:8]: yep

**Review by devsquad-mathews-lima — APPROVED** [review]
